### PR TITLE
Webterminal: Added SSL support with existing LE certificates.

### LINKED ISSUE #74: Webterminal Hassbian Script Over HTTPS
Webterminal Script for Hassbian is great but it does not work over HTTPS 
The script uses shellinabox and that has support for ssl  ... anyone ??

### PR DESCRIPTION
**Pull request by ludeeus**
## Description:
This add the option to use existing Let's Encrypt certificates.

Tested with:
- [X] Generated with Dehydrated. (DuckDNS script) tested by @ludeeus
- [X] Generated with certbot. _Simulated testing bt @ludeeus, se comment, this should work._

**Related issue (if applicable):** Fixes #74 

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Script has validation check of the job.
  
### If pertinent:
  - [X] Created/Updated documentation at `/docs`
  
image

### CODE REVIEW

**Comment by ludeeus** [comment]
@juan11perez Can you test this with your certificates generated with the certbot?

File: docs/webterminal.md
Comment (Landrash) [inline docs/webterminal.md]: Use **the same** domain.

File: package/opt/hassbian/suites/files/webterminalsslhelper.sh
Comment (Landrash) [inline package/opt/hassbian/suites/files/webterminalsslhelper.sh]: Doing it this way would not link the cert but only "copy" it. Preferable would be to create a symlink to the certificate so that it's linked to the used certificate.

**Review by Landrash — CHANGES_REQUESTED** [review]
Question is if we want to copy the certificate or if you should try to link the certificate instead.

File: package/opt/hassbian/suites/files/webterminalsslhelper.sh
Comment (ludeeus) [reply package/opt/hassbian/suites/files/webterminalsslhelper.sh]: In every guide I have found, and my own testing. the cert file used for shellinabox has to include both certificate and privkey, is it possible to merge with linking?

Comment (Landrash) [reply package/opt/hassbian/suites/files/webterminalsslhelper.sh]: I missed that. That would require some strange magic.
Most likely this will stop working as soon as the certificate expires. If your ok with that then we can merge this.
Could you add to the update function to update the certificate if it's there?

Comment (ludeeus) [reply package/opt/hassbian/suites/files/webterminalsslhelper.sh]: I hope not :( that defeats the purpose of the separate webterminalhelper.sh script..
Should't [this](https://github.com/home-assistant/hassbian-scripts/pull/130/files#diff-f9e7a0ede4e3162455ffa1299a2f1a19R52) update the cert for shellinabox 1AM every night?
looking at the timestamp after running the file it looks like that would work.
image
I can add a note about running that file in the docs?

"Could you add to the update function to update the certificate if it's there?"
Do you mean the update for dehydrated with the DuckDNS script, then no. that will result in certbot certs not being updated.
If this was not what you ment, enlighten me :D

Comment (Landrash) [reply package/opt/hassbian/suites/files/webterminalsslhelper.sh]: I missed that part of the webterminalhepler.sh script.

This look great now!

**Review by Landrash — DISMISSED** [review]
Looks good to me. 
Ready to merge? If so remove the WIP tag :)

**Comment by ludeeus** [comment]
I still want someone that can use certbot certs to verify this before er merge.

**Comment by juan11perez** [comment]
@ludeeus  i'll pull out my Pi and test it. I still have a valid installation, I think ;-)
where do I get the script from ?

**Comment by ludeeus** [comment]
@juan11perez Great!
Its in this PR, but hit me up on discord and I can help you :)

**Comment by juan11perez** [comment]
@ludeeus ok, give me about half hour and ill contact you in discord

**Comment by ludeeus** [comment]
Steps for testing this:
First upgrade hassbian-config to dev:
```bash
sudo hassbian-config upgrade hassbian-script && sudo hassbian-config upgrade hassbian-script-dev
```
Then download the new files:
```bash
sudo curl -o /opt/hassbian/suites/webterminal.sh https://raw.githubusercontent.com/ludeeus/hassbian-scripts/webterminal-ssl/package/opt/hassbian/suites/webterminal.sh && sudo curl -o /opt/hassbian/suites/files/webterminalhelper.sh https://raw.githubusercontent.com/ludeeus/hassbian-scripts/webterminal-ssl/package/opt/hassbian/suites/files/webterminalsslhelper.sh
```
Then run it:
```bash
sudo hassbian-config install webterminal --force
```

If you at an earlier time have installed webterminal, you need to change the settings for it:
`sudo nano /etc/default/shellinabox`
Change `SHELLINABOX_ARGS="--no-beep --disable-ssl"` to `SHELLINABOX_ARGS="--no-beep"`

**Review by Landrash — DISMISSED** [review]

**Comment by juan11perez** [comment]
@ludeeus not working. see entire process
https://hastebin.com/xuwajoheci.erl

**Comment by ludeeus** [comment]
I added my fulchain.pem and privkey.pem to an folder named `/etc/letsencrypt/live/mydomain.com/`
It ran the script and used those cert files.
This make me belive that this will work with certbot certs also.
@Landrash I have now removed WIP, can you take a check on this now? :D

**Review by Landrash — APPROVED** [review]
Looks good to me.
Great job!

**Pull request merged**
